token/transfer-hook/cli/src/meta.rs

```diff
@@ -0,0 +1,318 @@
+use {
+    serde::{Deserialize, Serialize},
+    solana_sdk::pubkey::Pubkey,
+    spl_tlv_account_resolution::{account::ExtraAccountMeta, seeds::Seed},
+    std::{fmt, path::Path, str::FromStr},
+    strum_macros::{EnumString, IntoStaticStr},
+};
+
+#[derive(Debug, Clone, Copy, PartialEq, EnumString, IntoStaticStr, Serialize, Deserialize)]
+#[serde(rename_all = "kebab-case")]
+#[strum(serialize_all = "kebab-case")]
+enum AccountMetaRole {
+    Readonly,
+    Writable,
+    ReadonlySigner,
+    WritableSigner,
+}
+impl AccountMetaRole {
+    fn bools(&self) -> (/* is_signer */ bool, /* is_writable */ bool) {
```

Comment:
nit: since this is pretty easy to use incorrectly, let's make this self-documenting with a struct like `AccountMetaRoleFlags` with `is_signer` and `is_writable` bool fields